lv_objx/lv_chart.c

```diff
@@ -387,6 +399,53 @@ void lv_chart_set_next(lv_obj_t * chart, lv_chart_series_t * ser, lv_coord_t y)
     lv_chart_refresh(chart);
 }
 
+/**
+ * Set the x-axis label of a chart
+ * @param chart pointer to a chart object
+ * @param text  pointer to the label text (not copied)
+ */
+void lv_chart_set_label(lv_obj_t* chart, const char* text)
+{
+    lv_chart_ext_t * ext = lv_obj_get_ext_attr(chart);
+    ext->x_axis.label = text;
+    lv_obj_refresh_ext_size(chart);
+}
+
+/**
+ * Set the x/y-axis ticks of a chart
+ * @param chart 			pointer to a chart object
+ * @param list_of_values 	list of string values, terminated with \n, except the last
+ * @param num_tick_marks 	if list_of_values is NULL: total number of ticks per axis
+ * 							else step in ticks between two value labels
+ * @param label				label to show for this axis (only X)
+ * @param options			extra options
+ */
+void lv_chart_set_x_ticks(	lv_obj_t* chart,
+							const char* list_of_values,
+							uint8_t num_tick_marks,
+							const char* label,
+							lv_chart_axis_options_t options)
+{
+    lv_chart_ext_t * ext = lv_obj_get_ext_attr(chart);
+    ext->x_axis.num_tick_marks = num_tick_marks;
+    ext->x_axis.list_of_values = list_of_values;
+    ext->x_axis.label = label;
+    ext->x_axis.options = options;
+}
+
+void lv_chart_set_y_ticks(	lv_obj_t* chart,
+							const char* list_of_values,
+							uint8_t num_tick_marks,
+							const char* label,
+							lv_chart_axis_options_t options)
+{
+    lv_chart_ext_t * ext = lv_obj_get_ext_attr(chart);
+    ext->y_axis.num_tick_marks = num_tick_marks;
+    ext->y_axis.list_of_values = list_of_values;
+    ext->y_axis.label = label;
```

Comment:
If only X label is displayed then it shouldn't be in the axis struct.  Maybe rename `x_axis.label` to `title`?

Reply:
Right now it is like title, you are right, but it is possible to add Y label in the future, I just don't have time and need to do it. This is the reason to leave it like this.

Reply:
I like your solution to name the axis by setting the first/last scale "number" to a "text".

As I'm thinking about it I'm not sure the title is required at all because:
- using an `lv_label` instead is easy and gives more flexibility. Obviously, it needs an extra object.
- I assume that the title will be used rarely so it makes the API more complicated (and even a little bit asymmetric now).

So I vote for removing `x/y_axis.label` completely. What do you think?

Reply:
It is ok to remove the label, I will do it. If the user needs label it can create separate object _lv_label_ and style and align it to the desired axis.